src/vs/workbench/contrib/notebook/browser/notebookOptions.ts

```diff
@@ -152,7 +156,11 @@ export class NotebookOptions extends Disposable {
 		// const { bottomToolbarGap, bottomToolbarHeight } = this._computeBottomToolbarDimensions(compactView, insertToolbarPosition, insertToolbarAlignment);
 		const fontSize = this.configurationService.getValue<number>('editor.fontSize');
 		const markupFontSize = this.configurationService.getValue<number>(NotebookSetting.markupFontSize);
-		const editorOptionsCustomizations = this.configurationService.getValue(NotebookSetting.cellEditorOptionsCustomizations);
+		const editorOptionsCustomizations = this.configurationService.getValue<{
+			'editor.indentSize': 'tabSize' | number;
+			'editor.tabSize': number;
+			'editor.insertSpaces': boolean;
+		}>(NotebookSetting.cellEditorOptionsCustomizations);
```

Comment:
This is actually trading one unsafe approach (any) for another (assuming user content is correct format). While this is certainly an improvement, the safest thing would be to pull in the value as `unknown` and validate/refine the types.

Reply:
Thanks, @Yoyokrazy I leave that improvement to you.

> ading one unsafe approach (any) for another (assuming user content is correct format)

Not sure what you mean here, this is a user setting, I do not see us using `unknown` anywhere when getting configuration values. or have I misunderstood your suggestion.
One change I'm making is to make these settings optional.

Please see the new change and whether they are sufficient,

Reply:
It's a user-controlled json file, there's nothing stopping a user from using `"editor.tabSize": null` or some other invalid value (which definitely happens), this would then likely lead to an exception if you trust the type. FWIW we don't validate do all over the codebase, but ideally we'd check if it's an object, if it has an `'editor'` key, if it's not null, if it's type is number, etc.

Reply:
>  user from using "editor.tabSize": null or some other invalid value (which definitely happens),

Turned into partial

> FWIW we don't validate do all over the codebase, but ideally we'd check if it's an object,

Yes, in fact a few lines above we have an object just like the same code I modified. hence my original question
Modified to check if object.

Reply:
Honestly i would have expected the configuration service to ignore data that does not conform to the schema, i.e. if user has `'editor.tabSize' = {some json object}` i would have thought the configuration service would return the default value.

Else its fairly inconsistent to validate in some places and not others.

Anyways, thats neither here nor there

Reply:
Fixed